hrcase/CRM/HRCase/Upgrader.php

```diff
@@ -329,10 +341,12 @@ private function up1402_removeUnusedCaseTypes() {
    * Creates/Updates default case types to be shipped
    * with this extension
    *
+   * WARNING: Resets activity types for an assignment, discarding customization
```

Comment:
ops I just relized this ,  I think you can just change pass a flag to this method :

```php
function createOrUpdateDefaultCaseTypes($defaultCaseTypes, $doNotUpdate = false )
```

and inside the method do : 

```php
      if (!empty($type['id'])) {
        $caseType['id'] = $type['id'];
      }
      
      if ((!empty( $caseType['id']) && !$doNotUpdate) || empty( $caseType['id'])) {
        civicrm_api3('CaseType', 'create', $caseType);
      }
```

and in your new upgrader you can pass (true) to it : 

`$this->createOrUpdateDefaultCaseTypes($defaultTypes, true);`